tests/lib/snaps/store/test-snapd-gpio-chardev/bin/service.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+if [ ! -d "$SNAP_COMMON/gpiochips" ]; then
+    echo "no chips found under $SNAP_COMMON/gpiochips, exiting..."
+    exit 0
+fi
+
+for gpiochip_file in "$SNAP_COMMON"/gpiochips/*; do
```

Comment:
You may want to consider `set nullglob`

From bash(1):
> nullglob
> If set, bash allows patterns which match no files (see Pathname Expansion above) to expand to a null string, rather than themselves.

This way you can drop the check above and just use the glob here.

Reply:
I am actually using the `no chips found under....` output in the spread test to check that services run normally even if they are unconnected.